src/watch.ts

```diff
@@ -45,25 +54,35 @@ export function useWatch({
 	const client = useClient()
 	const ops = useOnlyOps()
 
-	const [grid, setGrid] = useState<HGrid>(new HGrid())
+	const data = useRef<WatchData>()
 
-	const [isLoading, setIsLoading] = useState(true)
+	const watchData =
+		data.current ??
+		(data.current = {
```

Comment:
Is there a specific reason not to have this as the `initialValue` parameter of `useRef`?

Reply:
See other answer.